lib/packwerk/reference_to_offense/reference_checker.rb

```diff
@@ -0,0 +1,33 @@
+# typed: true
+# frozen_string_literal: true
+
+module Packwerk
+  module ReferenceToOffense
+    class ReferenceChecker
```

Comment:
It's a bit confusing that `ReferenceChecker` has a very similar name to `PrivacyChecker` and `DependencyChecker` yet is not a `Checker`.

I can't think of a good alternative name right now though...